src/license.jl

```diff
@@ -3,114 +3,77 @@
 # Use of this source code is governed by an MIT-style license that can be found
 # in the LICENSE.md file or at https://opensource.org/licenses/MIT.
 
-# lic checking file
-# -----------------
-
-# license check empty function
-# ----------------------------
-function emptyliccheck(lic::Vector{Cint})
-    return lic
-end
-
-function touchlic(path)
-    f = open(path)
-    return close(f)
-end
-
 function get_xpauthpath(xpauth_path = "", verbose::Bool = true)
     XPAUTH = "xpauth.xpr"
-
-    candidates = []
-
-    # user sent the complete path
-    push!(candidates, xpauth_path)
-
-    # user sent directory
-    push!(candidates, joinpath(xpauth_path, XPAUTH))
-
-    # default env (not metioned in manual)
-    if haskey(ENV, "XPAUTH_PATH")
-        xpauth_path = replace(ENV["XPAUTH_PATH"], "\"" => "")
-        push!(candidates, joinpath(xpauth_path, XPAUTH))
-    end
-
-    # default lib dir
-    if haskey(ENV, "XPRESSDIR")
-        xpressdir = replace(ENV["XPRESSDIR"], "\"" => "")
-        push!(candidates, joinpath(xpressdir, "bin", XPAUTH))
+    candidates = [xpauth_path, joinpath(xpauth_path, XPAUTH)]
+    for key in ("XPAUTH_PATH", "XPRESSDIR")
+        if haskey(ENV, key)
+            push!(candidates, joinpath(replace(ENV[key], "\"" => ""), XPAUTH))
+        end
     end
-
-    # user´s lib dir
     push!(candidates, joinpath(dirname(dirname(libxprs)), "bin", XPAUTH))
-
-    for i in candidates
-        if isfile(i)
+    for candidate in candidates
+        if isfile(candidate)
             if verbose && !haskey(ENV, "XPRESS_JL_NO_INFO")
-                @info("Xpress: Found license file $i")
+                @info("Xpress: Found license file $candidate")
             end
-            return i
+            return candidate
         end
     end
-
     return error(
-        "Could not find xpauth.xpr license file. Check XPRESSDIR or XPAUTH_PATH environment variables.",
+        "Could not find xpauth.xpr license file. Set the `XPRESSDIR` or " *
+        "`XPAUTH_PATH` environment variables.",
     )
 end
+
 """
-    userlic(; liccheck::Function = emptyliccheck, xpauth_path::String = "" )
-Performs license chhecking with `liccheck` validation function on dir `xpauth_path`
+    userlic(;
+        liccheck::Function = identity,
+        verbose::Bool = true,
+        xpauth_path::String = ""
+    )
+
+Performs license chhecking with `liccheck` validation function on dir
+`xpauth_path`
 """
 function userlic(;
+    liccheck::Function = identity,
     verbose::Bool = true,
-    liccheck::Function = emptyliccheck,
     xpauth_path::String = "",
 )
-
+    verbose &= !haskey(ENV, "XPRESS_JL_NO_INFO")
     # change directory to reach all libs
     initdir = pwd()
     if isdir(dirname(libxprs))
         cd(dirname(libxprs))
     end
```

Comment:
Why are we cd'ing? Just so that people can passing an `xpauth_path` that is relative to the directory of their library?